<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Single Sample Directional Gene Set Analysis (ssdGSA)</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Single Sample Directional Gene Set Analysis (ssdGSA)</h1>



<p> </p>
<div id="load-package-for-use" class="section level2">
<h2>Load package for use</h2>
<p> </p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ssdGSA)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span></code></pre></div>
<p> </p>
</div>
<div id="load-example-data-sets" class="section level2">
<h2>Load Example Data Sets</h2>
<p> </p>
<p>Let’s first load an example data matrix which contains gene expression information with ENTREZ ID as its row names. For the provided sample data matrix in this package, we only have ten samples in total.  </p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Data_matrix <span class="ot">&lt;-</span> ssdGSA<span class="sc">::</span>data_matrix_entrezID</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">round</span>(Data_matrix, <span class="dv">3</span>)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">S1</th>
<th align="right">S2</th>
<th align="right">S3</th>
<th align="right">S4</th>
<th align="right">S5</th>
<th align="right">S6</th>
<th align="right">S7</th>
<th align="right">S8</th>
<th align="right">S9</th>
<th align="right">S10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2268</td>
<td align="right">4.429</td>
<td align="right">3.761</td>
<td align="right">3.706</td>
<td align="right">4.760</td>
<td align="right">2.680</td>
<td align="right">3.937</td>
<td align="right">4.397</td>
<td align="right">4.278</td>
<td align="right">3.831</td>
<td align="right">3.844</td>
</tr>
<tr class="even">
<td align="left">572</td>
<td align="right">4.616</td>
<td align="right">4.760</td>
<td align="right">4.944</td>
<td align="right">4.497</td>
<td align="right">4.573</td>
<td align="right">4.575</td>
<td align="right">4.709</td>
<td align="right">5.152</td>
<td align="right">5.142</td>
<td align="right">4.936</td>
</tr>
<tr class="odd">
<td align="left">952</td>
<td align="right">2.359</td>
<td align="right">0.311</td>
<td align="right">1.894</td>
<td align="right">3.600</td>
<td align="right">-0.438</td>
<td align="right">2.163</td>
<td align="right">3.499</td>
<td align="right">1.738</td>
<td align="right">-0.415</td>
<td align="right">2.279</td>
</tr>
<tr class="even">
<td align="left">292</td>
<td align="right">7.752</td>
<td align="right">8.537</td>
<td align="right">8.201</td>
<td align="right">7.817</td>
<td align="right">7.530</td>
<td align="right">7.964</td>
<td align="right">8.026</td>
<td align="right">7.926</td>
<td align="right">8.287</td>
<td align="right">7.682</td>
</tr>
<tr class="odd">
<td align="left">9020</td>
<td align="right">5.614</td>
<td align="right">5.109</td>
<td align="right">5.317</td>
<td align="right">5.598</td>
<td align="right">5.300</td>
<td align="right">5.241</td>
<td align="right">5.264</td>
<td align="right">5.449</td>
<td align="right">5.157</td>
<td align="right">5.477</td>
</tr>
<tr class="even">
<td align="left">6376</td>
<td align="right">5.920</td>
<td align="right">5.540</td>
<td align="right">5.005</td>
<td align="right">6.520</td>
<td align="right">5.321</td>
<td align="right">5.629</td>
<td align="right">5.904</td>
<td align="right">5.381</td>
<td align="right">5.165</td>
<td align="right">5.610</td>
</tr>
</tbody>
</table>
<p> </p>
<p>Alternatively, we can load a data matrix which has ENSEMBL ID as its row names, such as the following example data matrix.  </p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">round</span>(ssdGSA<span class="sc">::</span>data_matrix, <span class="dv">3</span>)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">S1</th>
<th align="right">S2</th>
<th align="right">S3</th>
<th align="right">S4</th>
<th align="right">S5</th>
<th align="right">S6</th>
<th align="right">S7</th>
<th align="right">S8</th>
<th align="right">S9</th>
<th align="right">S10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000938</td>
<td align="right">4.429</td>
<td align="right">3.761</td>
<td align="right">3.706</td>
<td align="right">4.760</td>
<td align="right">2.680</td>
<td align="right">3.937</td>
<td align="right">4.397</td>
<td align="right">4.278</td>
<td align="right">3.831</td>
<td align="right">3.844</td>
</tr>
<tr class="even">
<td align="left">ENSG00000002330</td>
<td align="right">4.616</td>
<td align="right">4.760</td>
<td align="right">4.944</td>
<td align="right">4.497</td>
<td align="right">4.573</td>
<td align="right">4.575</td>
<td align="right">4.709</td>
<td align="right">5.152</td>
<td align="right">5.142</td>
<td align="right">4.936</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000004468</td>
<td align="right">2.359</td>
<td align="right">0.311</td>
<td align="right">1.894</td>
<td align="right">3.600</td>
<td align="right">-0.438</td>
<td align="right">2.163</td>
<td align="right">3.499</td>
<td align="right">1.738</td>
<td align="right">-0.415</td>
<td align="right">2.279</td>
</tr>
<tr class="even">
<td align="left">ENSG00000005022</td>
<td align="right">7.752</td>
<td align="right">8.537</td>
<td align="right">8.201</td>
<td align="right">7.817</td>
<td align="right">7.530</td>
<td align="right">7.964</td>
<td align="right">8.026</td>
<td align="right">7.926</td>
<td align="right">8.287</td>
<td align="right">7.682</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000006062</td>
<td align="right">5.614</td>
<td align="right">5.109</td>
<td align="right">5.317</td>
<td align="right">5.598</td>
<td align="right">5.300</td>
<td align="right">5.241</td>
<td align="right">5.264</td>
<td align="right">5.449</td>
<td align="right">5.157</td>
<td align="right">5.477</td>
</tr>
<tr class="even">
<td align="left">ENSG00000006210</td>
<td align="right">5.920</td>
<td align="right">5.540</td>
<td align="right">5.005</td>
<td align="right">6.520</td>
<td align="right">5.321</td>
<td align="right">5.629</td>
<td align="right">5.904</td>
<td align="right">5.381</td>
<td align="right">5.165</td>
<td align="right">5.610</td>
</tr>
</tbody>
</table>
<p> </p>
<p>We can use the function <code>transform_ensembl_2_entrez</code> in the proposed <code>ssdGSA</code> package to transform ENSEMBL ID into ENTREZ ID.  </p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>Data_matrix <span class="ot">&lt;-</span> <span class="fu">transform_ensembl_2_entrez</span>(ssdGSA<span class="sc">::</span>data_matrix)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">round</span>(Data_matrix, <span class="dv">3</span>)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">S1</th>
<th align="right">S2</th>
<th align="right">S3</th>
<th align="right">S4</th>
<th align="right">S5</th>
<th align="right">S6</th>
<th align="right">S7</th>
<th align="right">S8</th>
<th align="right">S9</th>
<th align="right">S10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2268</td>
<td align="right">4.429</td>
<td align="right">3.761</td>
<td align="right">3.706</td>
<td align="right">4.760</td>
<td align="right">2.680</td>
<td align="right">3.937</td>
<td align="right">4.397</td>
<td align="right">4.278</td>
<td align="right">3.831</td>
<td align="right">3.844</td>
</tr>
<tr class="even">
<td align="left">572</td>
<td align="right">4.616</td>
<td align="right">4.760</td>
<td align="right">4.944</td>
<td align="right">4.497</td>
<td align="right">4.573</td>
<td align="right">4.575</td>
<td align="right">4.709</td>
<td align="right">5.152</td>
<td align="right">5.142</td>
<td align="right">4.936</td>
</tr>
<tr class="odd">
<td align="left">952</td>
<td align="right">2.359</td>
<td align="right">0.311</td>
<td align="right">1.894</td>
<td align="right">3.600</td>
<td align="right">-0.438</td>
<td align="right">2.163</td>
<td align="right">3.499</td>
<td align="right">1.738</td>
<td align="right">-0.415</td>
<td align="right">2.279</td>
</tr>
<tr class="even">
<td align="left">292</td>
<td align="right">7.752</td>
<td align="right">8.537</td>
<td align="right">8.201</td>
<td align="right">7.817</td>
<td align="right">7.530</td>
<td align="right">7.964</td>
<td align="right">8.026</td>
<td align="right">7.926</td>
<td align="right">8.287</td>
<td align="right">7.682</td>
</tr>
<tr class="odd">
<td align="left">9020</td>
<td align="right">5.614</td>
<td align="right">5.109</td>
<td align="right">5.317</td>
<td align="right">5.598</td>
<td align="right">5.300</td>
<td align="right">5.241</td>
<td align="right">5.264</td>
<td align="right">5.449</td>
<td align="right">5.157</td>
<td align="right">5.477</td>
</tr>
<tr class="even">
<td align="left">6376</td>
<td align="right">5.920</td>
<td align="right">5.540</td>
<td align="right">5.005</td>
<td align="right">6.520</td>
<td align="right">5.321</td>
<td align="right">5.629</td>
<td align="right">5.904</td>
<td align="right">5.381</td>
<td align="right">5.165</td>
<td align="right">5.610</td>
</tr>
</tbody>
</table>
<p> </p>
<p>Now the data matrix has ENTREZ ID as its row names. Please make sure that the data matrix you input in the main function <code>ssdGSA</code> and <code>ssdGSA_individual</code> has gene ENTREZ ID as its row names.    </p>
<p>Then, let’s load an example gene sets, which is a list containing gene ENTREZ ID in each gene set. In this example gene sets, there are in total 10 gene sets (corresponding to 10 pathways in this case), and we display 3 gene sets.  </p>
<pre><code>#&gt; $TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY
#&gt;   [1] &quot;7535&quot;  &quot;3725&quot;  &quot;10892&quot; &quot;5163&quot;  &quot;940&quot;   &quot;5535&quot;  &quot;5533&quot;  &quot;5534&quot;  &quot;10298&quot;
#&gt;  [10] &quot;29851&quot; &quot;7409&quot;  &quot;3551&quot;  &quot;1326&quot;  &quot;6885&quot;  &quot;207&quot;   &quot;208&quot;   &quot;4893&quot;  &quot;10000&quot;
#&gt;  [19] &quot;7410&quot;  &quot;63928&quot; &quot;10725&quot; &quot;4790&quot;  &quot;4793&quot;  &quot;4792&quot;  &quot;56924&quot; &quot;1493&quot;  &quot;4690&quot; 
#&gt;  [28] &quot;4794&quot;  &quot;23533&quot; &quot;1437&quot;  &quot;920&quot;   &quot;4776&quot;  &quot;1432&quot;  &quot;6300&quot;  &quot;5970&quot;  &quot;4775&quot; 
#&gt;  [37] &quot;3845&quot;  &quot;925&quot;   &quot;926&quot;   &quot;919&quot;   &quot;8440&quot;  &quot;9020&quot;  &quot;57144&quot; &quot;915&quot;   &quot;5058&quot; 
#&gt;  [46] &quot;916&quot;   &quot;917&quot;   &quot;23624&quot; &quot;3937&quot;  &quot;5335&quot;  &quot;4773&quot;  &quot;4772&quot;  &quot;3932&quot;  &quot;998&quot;  
#&gt;  [55] &quot;10125&quot; &quot;6655&quot;  &quot;5063&quot;  &quot;1147&quot;  &quot;5894&quot;  &quot;7124&quot;  &quot;84433&quot; &quot;5062&quot;  &quot;387&quot;  
#&gt;  [64] &quot;5777&quot;  &quot;5133&quot;  &quot;5601&quot;  &quot;5600&quot;  &quot;5605&quot;  &quot;5588&quot;  &quot;5603&quot;  &quot;5604&quot;  &quot;5609&quot; 
#&gt;  [73] &quot;9402&quot;  &quot;2885&quot;  &quot;5595&quot;  &quot;3586&quot;  &quot;5788&quot;  &quot;6654&quot;  &quot;5594&quot;  &quot;3265&quot;  &quot;8517&quot; 
#&gt;  [82] &quot;7006&quot;  &quot;2932&quot;  &quot;868&quot;   &quot;867&quot;   &quot;3702&quot;  &quot;27040&quot; &quot;3458&quot;  &quot;3558&quot;  &quot;959&quot;  
#&gt;  [91] &quot;8503&quot;  &quot;1019&quot;  &quot;5532&quot;  &quot;1739&quot;  &quot;5530&quot;  &quot;5290&quot;  &quot;5291&quot;  &quot;5293&quot;  &quot;8915&quot; 
#&gt; [100] &quot;2534&quot;  &quot;3565&quot;  &quot;2353&quot;  &quot;3567&quot;  &quot;5294&quot;  &quot;5295&quot;  &quot;10451&quot; &quot;5296&quot;  &quot;11261&quot;
#&gt; 
#&gt; $TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY
#&gt;   [1] &quot;3725&quot;   &quot;941&quot;    &quot;942&quot;    &quot;6772&quot;   &quot;54106&quot;  &quot;6348&quot;   &quot;6416&quot;   &quot;6352&quot;  
#&gt;   [9] &quot;4615&quot;   &quot;6351&quot;   &quot;3929&quot;   &quot;8737&quot;   &quot;10333&quot;  &quot;3551&quot;   &quot;23643&quot;  &quot;1326&quot;  
#&gt;  [17] &quot;6885&quot;   &quot;207&quot;    &quot;208&quot;    &quot;929&quot;    &quot;1513&quot;   &quot;10000&quot;  &quot;6696&quot;   &quot;9641&quot;  
#&gt;  [25] &quot;4790&quot;   &quot;4792&quot;   &quot;23533&quot;  &quot;7100&quot;   &quot;114609&quot; &quot;1432&quot;   &quot;6300&quot;   &quot;5970&quot;  
#&gt;  [33] &quot;7189&quot;   &quot;7187&quot;   &quot;54472&quot;  &quot;6373&quot;   &quot;23118&quot;  &quot;51284&quot;  &quot;51311&quot;  &quot;841&quot;   
#&gt;  [41] &quot;3627&quot;   &quot;1147&quot;   &quot;148022&quot; &quot;7124&quot;   &quot;3442&quot;   &quot;3441&quot;   &quot;3440&quot;   &quot;8772&quot;  
#&gt;  [49] &quot;3439&quot;   &quot;51135&quot;  &quot;3576&quot;   &quot;5601&quot;   &quot;3593&quot;   &quot;5602&quot;   &quot;5600&quot;   &quot;5605&quot;  
#&gt;  [57] &quot;5606&quot;   &quot;3592&quot;   &quot;5603&quot;   &quot;5604&quot;   &quot;5609&quot;   &quot;5608&quot;   &quot;3451&quot;   &quot;3452&quot;  
#&gt;  [65] &quot;3443&quot;   &quot;3444&quot;   &quot;3445&quot;   &quot;3446&quot;   &quot;3447&quot;   &quot;3448&quot;   &quot;3449&quot;   &quot;5595&quot;  
#&gt;  [73] &quot;5879&quot;   &quot;5594&quot;   &quot;3661&quot;   &quot;8517&quot;   &quot;5599&quot;   &quot;3456&quot;   &quot;3454&quot;   &quot;3455&quot;  
#&gt;  [81] &quot;353376&quot; &quot;3553&quot;   &quot;3654&quot;   &quot;958&quot;    &quot;8503&quot;   &quot;29110&quot;  &quot;5290&quot;   &quot;4283&quot;  
#&gt;  [89] &quot;5291&quot;   &quot;5293&quot;   &quot;7098&quot;   &quot;7099&quot;   &quot;10454&quot;  &quot;7096&quot;   &quot;2353&quot;   &quot;7097&quot;  
#&gt;  [97] &quot;5294&quot;   &quot;3663&quot;   &quot;5295&quot;   &quot;3569&quot;   &quot;5296&quot;   &quot;3665&quot;  
#&gt; 
#&gt; $NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY
#&gt;   [1] &quot;3725&quot;   &quot;10818&quot;  &quot;805&quot;    &quot;5163&quot;   &quot;801&quot;    &quot;27330&quot;  &quot;7161&quot;   &quot;4217&quot;  
#&gt;   [9] &quot;4214&quot;   &quot;673&quot;    &quot;9252&quot;   &quot;4215&quot;   &quot;3551&quot;   &quot;25&quot;     &quot;207&quot;    &quot;399694&quot;
#&gt;  [17] &quot;208&quot;    &quot;4893&quot;   &quot;10000&quot;  &quot;25970&quot;  &quot;4790&quot;   &quot;4793&quot;   &quot;4792&quot;   &quot;4794&quot;  
#&gt;  [25] &quot;23533&quot;  &quot;27018&quot;  &quot;1432&quot;   &quot;6300&quot;   &quot;5970&quot;   &quot;3845&quot;   &quot;53358&quot;  &quot;7189&quot;  
#&gt;  [33] &quot;51806&quot;  &quot;4803&quot;   &quot;627&quot;    &quot;4804&quot;   &quot;10019&quot;  &quot;1445&quot;   &quot;596&quot;    &quot;7529&quot;  
#&gt;  [41] &quot;814&quot;    &quot;815&quot;    &quot;816&quot;    &quot;2309&quot;   &quot;817&quot;    &quot;818&quot;    &quot;8660&quot;   &quot;6272&quot;  
#&gt;  [49] &quot;810&quot;    &quot;1398&quot;   &quot;10603&quot;  &quot;1399&quot;   &quot;5335&quot;   &quot;5336&quot;   &quot;7534&quot;   &quot;396&quot;   
#&gt;  [57] &quot;808&quot;    &quot;7533&quot;   &quot;397&quot;    &quot;7532&quot;   &quot;7531&quot;   &quot;998&quot;    &quot;6655&quot;   &quot;5781&quot;  
#&gt;  [65] &quot;5894&quot;   &quot;6464&quot;   &quot;8767&quot;   &quot;387&quot;    &quot;572&quot;    &quot;51135&quot;  &quot;5580&quot;   &quot;5601&quot;  
#&gt;  [73] &quot;5602&quot;   &quot;9500&quot;   &quot;5600&quot;   &quot;5605&quot;   &quot;5603&quot;   &quot;5604&quot;   &quot;581&quot;    &quot;5609&quot;  
#&gt;  [81] &quot;5607&quot;   &quot;6196&quot;   &quot;2889&quot;   &quot;6197&quot;   &quot;2885&quot;   &quot;5595&quot;   &quot;8986&quot;   &quot;5598&quot;  
#&gt;  [89] &quot;57498&quot;  &quot;5879&quot;   &quot;6654&quot;   &quot;5594&quot;   &quot;6195&quot;   &quot;10971&quot;  &quot;3265&quot;   &quot;5599&quot;  
#&gt;  [97] &quot;11108&quot;  &quot;2932&quot;   &quot;9261&quot;   &quot;5663&quot;   &quot;4909&quot;   &quot;4908&quot;   &quot;3656&quot;   &quot;3654&quot;  
#&gt; [105] &quot;5906&quot;   &quot;8503&quot;   &quot;5908&quot;   &quot;11213&quot;  &quot;7157&quot;   &quot;5290&quot;   &quot;5291&quot;   &quot;5293&quot;  
#&gt; [113] &quot;2549&quot;   &quot;25759&quot;  &quot;356&quot;    &quot;468&quot;    &quot;3667&quot;   &quot;5294&quot;   &quot;4915&quot;   &quot;10782&quot; 
#&gt; [121] &quot;5295&quot;   &quot;4914&quot;   &quot;8471&quot;   &quot;5296&quot;   &quot;4916&quot;   &quot;163688&quot;</code></pre>
<p> </p>
<p>This is a list of gene sets with gene set names as component names, and each component is a vector of gene ENTREZ ID.</p>
<p> </p>
<p>Also, we need to load the direction matrix which contains directionality information for each gene from summary statistics.</p>
<table>
<thead>
<tr class="header">
<th align="right">gene</th>
<th align="right">ES</th>
<th align="right">pval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">10000</td>
<td align="right">0.200</td>
<td align="right">0.323</td>
</tr>
<tr class="even">
<td align="right">10019</td>
<td align="right">1.096</td>
<td align="right">0.272</td>
</tr>
<tr class="odd">
<td align="right">10125</td>
<td align="right">0.422</td>
<td align="right">0.264</td>
</tr>
<tr class="even">
<td align="right">1019</td>
<td align="right">-1.030</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="right">10235</td>
<td align="right">2.008</td>
<td align="right">0.067</td>
</tr>
<tr class="even">
<td align="right">1026</td>
<td align="right">0.884</td>
<td align="right">0.075</td>
</tr>
</tbody>
</table>
<p> </p>
<p>This direction matrix contains directionality information for each gene, such as effect size (ES), p value, false positive rate (FDR) from summary statistics. Each row of the matrix is for one gene, and there should be at least two columns (with the 1st column containing gene ENTREZ ID, and 2nd column containing directionality information of that gene). For the provided sample direction matrix in this package we include three columns: <code>gene</code>, <code>ES</code> and <code>pval</code>, corresponding to gene ENTREZ ID, effect size and p value. Also note that when direction matrix is missing, scores from traditional single sample gene set analysis would be calculated by the proposed <code>ssdGSA</code> package.  </p>
</div>
<div id="functions" class="section level2">
<h2>Functions</h2>
<p> </p>
<div id="ssdgsa" class="section level3">
<h3>ssdGSA</h3>
<p> </p>
<p>This function is to do single sample directional gene set analysis, which inherits the standard gene set variation analysis(GSVA) method, but also provides the option to use summary statistics from any analysis (disease vs healthy, LS vs NL, etc..) input to define the direction of gene sets used for directional gene set score calculation for a given disease. Note that this function is specific for using group weighted scores.    </p>
<p>Below are the default parameters for <code>ssdGSA</code>. You can change them to modify your output. Use <code>help(ssdGSA)</code> to learn more about the parameters.  </p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ssdGSA</span>(<span class="at">Data =</span> Data_matrix,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">Gene_sets =</span> Gene_sets,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">Direction_matrix =</span> Direction_matrix, </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">GSA_weight =</span> <span class="st">&quot;group_weighted&quot;</span>,</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">GSA_weighted_by =</span> <span class="st">&quot;sum.ES&quot;</span>, <span class="co">#options are: &quot;sum.ES&quot;, &quot;avg.ES&quot;, &quot;median.ES&quot;</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">GSA_method =</span> <span class="st">&quot;gsva&quot;</span>, <span class="co">#&quot;options are: &quot;gsva&quot;, &quot;ssgsea&quot;, &quot;zscore&quot;, &quot;avg.exprs&quot;, and &quot;median.exprs&quot;</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">min.sz =</span> <span class="dv">1</span>,  <span class="co"># GSVA parameter</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">max.sz =</span> <span class="dv">2000</span>, <span class="co"># GSVA parameter</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">mx.diff =</span> <span class="cn">TRUE</span> <span class="co"># GSVA parameter</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning messages: </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 100% (3/3) of gene sets have information missing in the data matrix for at least one gene; </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0% (0/3) of gene sets have information missing in the direction matrix for at least one gene.</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY: </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6.48% of genes (7/108) in this gene set have information missing in the data matrix; </span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY: </span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  18.63% of genes (19/102) in this gene set have information missing in the data matrix; </span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY: </span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2.38% of genes (3/126) in this gene set have information missing in the data matrix;</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Note: Running GSA for 3 gene sets using &#39;gsva&#39; method with &#39;group_weighted&#39; weights.</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating GSVA scores for 3 gene sets.</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating ECDFs with Gaussian kernels</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=======================                                               |  33%  |                                                                              |===============================================                       |  67%  |                                                                              |======================================================================| 100%</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating GSVA scores for 3 gene sets.</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating ECDFs with Gaussian kernels</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=======================                                               |  33%  |                                                                              |===============================================                       |  67%  |                                                                              |======================================================================| 100%</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                         S1         S2</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.8899907 -0.7563572</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.4402918 -0.7308813</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.6600825 -0.4376452</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                           S3        S4</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     -0.11595511 0.9147812</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY     0.02531708 0.8983102</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY  0.32165136 0.5579457</span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                          S5          S6</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     -0.8046286  0.67310666</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    -0.0860897 -0.21186968</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY -0.2392317  0.02336175</span></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                         S7         S8</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.5649402 -0.2881176</span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.7599040 -0.2073954</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.0519870 -0.3424723</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                          S9        S10</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     -0.8291629 -0.1599448</span></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    -1.0423261  0.3924671</span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY -0.7903744  0.1108400</span></span></code></pre></div>
<p> </p>
<p>A matrix of directional gene set scores from single sample directional gene set analysis when using group weighted scores, with rows corresponding to gene sets and columns corresponding to different samples is returned.  </p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ssdGSA</span>(<span class="at">Data =</span> Data_matrix,</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">Gene_sets =</span> Gene_sets,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">Direction_matrix =</span> <span class="cn">NULL</span>, </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">GSA_weight =</span> <span class="st">&quot;group_weighted&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">GSA_weighted_by =</span> <span class="st">&quot;sum.ES&quot;</span>, <span class="co">#options are: &quot;sum.ES&quot;, &quot;avg.ES&quot;, &quot;median.ES&quot;</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">GSA_method =</span> <span class="st">&quot;gsva&quot;</span>, <span class="co">#&quot;options are: &quot;gsva&quot;, &quot;ssgsea&quot;, &quot;zscore&quot;, &quot;avg.exprs&quot;, and &quot;median.exprs&quot;</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">min.sz =</span> <span class="dv">6</span>,  <span class="co"># GSVA parameter</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">max.sz =</span> <span class="dv">2000</span>, <span class="co"># GSVA parameter</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">mx.diff =</span> <span class="cn">TRUE</span> <span class="co"># GSVA parameter</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning messages: </span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 100% (3/3) of gene sets have information missing in the data matrix for at least one gene.</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY: </span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6.48% of genes (7/108) in this gene set have information missing in the data matrix; </span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY: </span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  18.63% of genes (19/102) in this gene set have information missing in the data matrix; </span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY: </span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2.38% of genes (3/126) in this gene set have information missing in the data matrix;</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Note: Running GSVA for 3 gene sets using &#39;gsva&#39; method with &#39;group_weighted&#39; weights.</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating GSVA scores for 3 gene sets.</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating ECDFs with Gaussian kernels</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=======================                                               |  33%  |                                                                              |===============================================                       |  67%  |                                                                              |======================================================================| 100%</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                          S1          S2</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.32529011 -0.06270977</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.03361398 -0.22037641</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.05229527  0.26714173</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                           S3          S4</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     -0.06606869  0.08399112</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY     0.02795322  0.24911101</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY -0.03985403 -0.17438679</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                          S5          S6</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.13796232  0.26365975</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.15094708 -0.06247332</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.03404076 -0.08262855</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                           S7          S8</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     -0.05860165 -0.20555643</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY     0.18322508 -0.08882073</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY -0.17181714 -0.10191413</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                          S9        S10</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     -0.2702770 -0.1182854</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    -0.3976409  0.1118144</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY -0.0553835  0.1801999</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;geneSets&quot;)</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;geneSets&quot;)$TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] &quot;7535&quot;  &quot;3725&quot;  &quot;10892&quot; &quot;5163&quot;  &quot;940&quot;   &quot;5533&quot;  &quot;5534&quot;  &quot;10298&quot; &quot;29851&quot;</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [10] &quot;7409&quot;  &quot;3551&quot;  &quot;1326&quot;  &quot;6885&quot;  &quot;207&quot;   &quot;208&quot;   &quot;4893&quot;  &quot;10000&quot; &quot;7410&quot; </span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [19] &quot;63928&quot; &quot;10725&quot; &quot;4790&quot;  &quot;4793&quot;  &quot;4792&quot;  &quot;56924&quot; &quot;1493&quot;  &quot;4690&quot;  &quot;4794&quot; </span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [28] &quot;23533&quot; &quot;920&quot;   &quot;4776&quot;  &quot;1432&quot;  &quot;6300&quot;  &quot;5970&quot;  &quot;4775&quot;  &quot;3845&quot;  &quot;925&quot;  </span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [37] &quot;926&quot;   &quot;919&quot;   &quot;8440&quot;  &quot;9020&quot;  &quot;915&quot;   &quot;5058&quot;  &quot;916&quot;   &quot;917&quot;   &quot;23624&quot;</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [46] &quot;3937&quot;  &quot;5335&quot;  &quot;4773&quot;  &quot;4772&quot;  &quot;3932&quot;  &quot;998&quot;   &quot;10125&quot; &quot;6655&quot;  &quot;5063&quot; </span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [55] &quot;1147&quot;  &quot;5894&quot;  &quot;7124&quot;  &quot;84433&quot; &quot;5062&quot;  &quot;387&quot;   &quot;5777&quot;  &quot;5133&quot;  &quot;5601&quot; </span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [64] &quot;5600&quot;  &quot;5605&quot;  &quot;5588&quot;  &quot;5603&quot;  &quot;5604&quot;  &quot;5609&quot;  &quot;9402&quot;  &quot;2885&quot;  &quot;5595&quot; </span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [73] &quot;3586&quot;  &quot;5788&quot;  &quot;6654&quot;  &quot;5594&quot;  &quot;3265&quot;  &quot;8517&quot;  &quot;7006&quot;  &quot;2932&quot;  &quot;868&quot;  </span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [82] &quot;867&quot;   &quot;3702&quot;  &quot;27040&quot; &quot;3458&quot;  &quot;959&quot;   &quot;8503&quot;  &quot;1019&quot;  &quot;5532&quot;  &quot;1739&quot; </span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [91] &quot;5530&quot;  &quot;5290&quot;  &quot;5291&quot;  &quot;5293&quot;  &quot;8915&quot;  &quot;2534&quot;  &quot;2353&quot;  &quot;5294&quot;  &quot;5295&quot; </span></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [100] &quot;10451&quot; &quot;11261&quot;</span></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;geneSets&quot;)$TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY</span></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;3725&quot;   &quot;941&quot;    &quot;942&quot;    &quot;6772&quot;   &quot;54106&quot;  &quot;6348&quot;   &quot;6416&quot;   &quot;6352&quot;  </span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;4615&quot;   &quot;6351&quot;   &quot;8737&quot;   &quot;10333&quot;  &quot;3551&quot;   &quot;23643&quot;  &quot;1326&quot;   &quot;6885&quot;  </span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;207&quot;    &quot;208&quot;    &quot;929&quot;    &quot;1513&quot;   &quot;10000&quot;  &quot;6696&quot;   &quot;9641&quot;   &quot;4790&quot;  </span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;4792&quot;   &quot;23533&quot;  &quot;7100&quot;   &quot;114609&quot; &quot;1432&quot;   &quot;6300&quot;   &quot;5970&quot;   &quot;7189&quot;  </span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] &quot;7187&quot;   &quot;54472&quot;  &quot;6373&quot;   &quot;23118&quot;  &quot;51284&quot;  &quot;51311&quot;  &quot;841&quot;    &quot;3627&quot;  </span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] &quot;1147&quot;   &quot;148022&quot; &quot;7124&quot;   &quot;8772&quot;   &quot;51135&quot;  &quot;3576&quot;   &quot;5601&quot;   &quot;5602&quot;  </span></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [49] &quot;5600&quot;   &quot;5605&quot;   &quot;5606&quot;   &quot;5603&quot;   &quot;5604&quot;   &quot;5609&quot;   &quot;5608&quot;   &quot;5595&quot;  </span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [57] &quot;5879&quot;   &quot;5594&quot;   &quot;3661&quot;   &quot;8517&quot;   &quot;5599&quot;   &quot;3454&quot;   &quot;3455&quot;   &quot;3553&quot;  </span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [65] &quot;3654&quot;   &quot;958&quot;    &quot;8503&quot;   &quot;29110&quot;  &quot;5290&quot;   &quot;4283&quot;   &quot;5291&quot;   &quot;5293&quot;  </span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [73] &quot;7098&quot;   &quot;7099&quot;   &quot;10454&quot;  &quot;7096&quot;   &quot;2353&quot;   &quot;7097&quot;   &quot;5294&quot;   &quot;3663&quot;  </span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [81] &quot;5295&quot;   &quot;3569&quot;   &quot;3665&quot;  </span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;geneSets&quot;)$NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY</span></span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] &quot;3725&quot;   &quot;10818&quot;  &quot;805&quot;    &quot;5163&quot;   &quot;801&quot;    &quot;27330&quot;  &quot;7161&quot;   &quot;4217&quot;  </span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [9] &quot;4214&quot;   &quot;673&quot;    &quot;9252&quot;   &quot;4215&quot;   &quot;3551&quot;   &quot;25&quot;     &quot;207&quot;    &quot;399694&quot;</span></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [17] &quot;208&quot;    &quot;4893&quot;   &quot;10000&quot;  &quot;25970&quot;  &quot;4790&quot;   &quot;4793&quot;   &quot;4792&quot;   &quot;4794&quot;  </span></span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [25] &quot;23533&quot;  &quot;27018&quot;  &quot;1432&quot;   &quot;6300&quot;   &quot;5970&quot;   &quot;3845&quot;   &quot;53358&quot;  &quot;7189&quot;  </span></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [33] &quot;51806&quot;  &quot;627&quot;    &quot;4804&quot;   &quot;10019&quot;  &quot;1445&quot;   &quot;596&quot;    &quot;7529&quot;   &quot;814&quot;   </span></span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [41] &quot;815&quot;    &quot;816&quot;    &quot;2309&quot;   &quot;817&quot;    &quot;818&quot;    &quot;8660&quot;   &quot;6272&quot;   &quot;810&quot;   </span></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [49] &quot;1398&quot;   &quot;10603&quot;  &quot;1399&quot;   &quot;5335&quot;   &quot;5336&quot;   &quot;7534&quot;   &quot;396&quot;    &quot;808&quot;   </span></span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [57] &quot;7533&quot;   &quot;397&quot;    &quot;7532&quot;   &quot;7531&quot;   &quot;998&quot;    &quot;6655&quot;   &quot;5781&quot;   &quot;5894&quot;  </span></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [65] &quot;6464&quot;   &quot;8767&quot;   &quot;387&quot;    &quot;572&quot;    &quot;51135&quot;  &quot;5580&quot;   &quot;5601&quot;   &quot;5602&quot;  </span></span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [73] &quot;9500&quot;   &quot;5600&quot;   &quot;5605&quot;   &quot;5603&quot;   &quot;5604&quot;   &quot;581&quot;    &quot;5609&quot;   &quot;5607&quot;  </span></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [81] &quot;6196&quot;   &quot;2889&quot;   &quot;6197&quot;   &quot;2885&quot;   &quot;5595&quot;   &quot;8986&quot;   &quot;5598&quot;   &quot;57498&quot; </span></span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [89] &quot;5879&quot;   &quot;6654&quot;   &quot;5594&quot;   &quot;6195&quot;   &quot;10971&quot;  &quot;3265&quot;   &quot;5599&quot;   &quot;11108&quot; </span></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [97] &quot;2932&quot;   &quot;9261&quot;   &quot;5663&quot;   &quot;4909&quot;   &quot;4908&quot;   &quot;3656&quot;   &quot;3654&quot;   &quot;5906&quot;  </span></span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [105] &quot;8503&quot;   &quot;5908&quot;   &quot;11213&quot;  &quot;7157&quot;   &quot;5290&quot;   &quot;5291&quot;   &quot;5293&quot;   &quot;2549&quot;  </span></span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [113] &quot;25759&quot;  &quot;356&quot;    &quot;468&quot;    &quot;3667&quot;   &quot;5294&quot;   &quot;4915&quot;   &quot;10782&quot;  &quot;5295&quot;  </span></span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [121] &quot;4914&quot;   &quot;4916&quot;   &quot;163688&quot;</span></span></code></pre></div>
<p> </p>
<p>Alternatively, when direction matrix is missing, i.e., <code>Direction_matrix = NULL</code>, scores from traditional single sample gene set analysis without directionality information will be calculated and returned.</p>
<p> </p>
</div>
<div id="ssdgsa_individual" class="section level3">
<h3>ssdGSA_individual</h3>
<p> </p>
<p>This function is to do single sample directional gene set analysis when using individual weighted scores.  </p>
<p>Below are the default parameters for <code>ssdGSA_individual</code>. You can change them to modify your output. Use <code>help(ssdGSA_individual)</code> to learn more about the parameters.  </p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ssdGSA_individual</span>(<span class="at">Data =</span> Data_matrix,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">Gene_sets =</span> Gene_sets,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">Direction_matrix =</span> Direction_matrix</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning messages: </span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 100% (3/3) of gene sets have information missing in the data matrix for at least one gene; </span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0% (0/3) of gene sets have information missing in the direction matrix for at least one gene.</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY: </span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6.48% of genes (7/108) in this gene set have information missing in the data matrix; </span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY: </span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  18.63% of genes (19/102) in this gene set have information missing in the data matrix; </span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY: </span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2.38% of genes (3/126) in this gene set have information missing in the data matrix;</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Note: Running GSA for 3 gene sets using individual weights.</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                         S1        S2        S3</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.8393467 0.7441525 0.7692775</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.9901754 0.7456950 0.9189434</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.5689334 0.5562217 0.5617713</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                         S4        S5        S6</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.8266904 0.6935935 0.8230364</span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.9890529 0.7882257 0.9169321</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.5752082 0.5474677 0.5705729</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                         S7        S8        S9</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.8144697 0.7539750 0.7184018</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    1.0088957 0.8382858 0.7202826</span></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.5698750 0.5602913 0.5584633</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                        S10</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY     0.7497758</span></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY    0.9084717</span></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 0.5641252</span></span></code></pre></div>
<p> </p>
<p>A matrix of directional gene set scores from single sample directional gene set analysis when using individual weighted scores, with rows corresponding to gene sets and columns corresponding to different samples is returned.    </p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
